tests/common.py

```diff
@@ -551,7 +551,9 @@ def verify_es5(self, filename):
   def build(self, filename, libraries=[], includes=[], force_c=False, js_outfile=True, emcc_args=[], output_basename=None):
     suffix = '.js' if js_outfile else '.wasm'
     compiler = [compiler_for(filename, force_c)]
-    if compiler[0] == EMCC:
+    if compiler[0] == EMCC and not self.get_setting('WASMFS'):
+      # TODO change test behaviour in the future when WASMFS becomes default file system
+      # WASMFS is excluded here since it currently requires stdlib++ functions
```

Comment:
This kind of worries me, to be honest. It seems like we might end up forgetting it, and then users would hit problems that don't occur in the test suite. That's a risk when working around issues in the test suite like this.

One possible idea is to move this to emcc.py, using the flag @sbc100 mentioned to link as cxx. We'll need that anyhow I think. However, I'm also ok with landing this if we put that on a TODO somewhere we are sure we won't forget.

Reply:
You mean using `-s DEFAULT_TO_CXX=0` here instead of `-nostdlib++`?  Sure we could try that.   I'll open a PR to do that.

Reply:
(Maybe what you meant, but to be sure I meant not here and instead in `emcc.py` - then it will work outside of the test suite too.)

Reply:
Are you suggesting we move forward with changing the default to `DEFAULT_TO_CXX=0` for all users?  This is the plan as layed out in https://github.com/emscripten-core/emscripten/issues/11121 but it could be a breaking change for many users and I'd like to start by issuing a warning.

The point of this line here is to opt into this new stricter behaviour within the test suite before we apply it to all users.

Reply:
No, sorry, I mean to add something like this in emcc.py:
```python
if WASMFS:
  DEFAULT_TO_CXX = 1
```
That would affect no one but a few tests atm.

Reply:
Yes that sounds like the correct solution.

Reply:
We can land this as is for now. I will keep the todo and add this setting in the following PR

Reply:
Followup PR sounds good.